### PR TITLE
Remove lots of 'inline's in IO.chpl for non-trivial procedures

### PR DESCRIPTION
**Pull request by bradcray**
While looking through the IO module this week, I was surprised
at the number of its routines marked as `inline`, particularly given
that I/O isn't a particularly low-overhead operation (where I'm
skeptical that the inlining was likely to help that much).

As a quick experiment, I tried pulling the `inline` keywords off
of any routine that seemed nontrivial by some definition.  In my
first very non-scientific run, this seemed to have a significant
impact on Arkouda's compilation time, but upon re-running, the
margin closed.  Using a more deterministic metric, the generated
code size only changed by a bit:

master: 1007426 lines of `*.c/h` code produced by `--savec output`
branch: 1001703

So that's only a ~0.5% difference.

Some of these large routines may have folded away to something
short by virtue of having conditionals or selects that are folded at
compile-time... but it still wasn't clear to me that they were obvious
candidates for inlining.

### CODE REVIEW

**Comment by bradcray** [comment]
Tagging @mppf since I think he wrote a lot of this code to begin with (though the inlines may or may not have been his), @e-kayrakli due to other recent IO compilation time improvements he made, and @ronawho who keeps pretty close track of Arkouda compilation times, in case any of you have comments/thoughts about this.

**Comment by mppf** [comment]
This is OK with me. I think I originally added them based on some performance experiments measuring I/O speeds (when all the data is already in the kernel's cache for filesystem data). We could revisit those experiments if we like.

**Comment by e-kayrakli** [comment]
Nice!

**Comment by ronawho** [comment]
Related to https://github.com/Cray/chapel-private/issues/1260 (just creating the auto-link so this is easier to find from there in the future)

**Comment by bradcray** [comment]
Hmm, it looks like I got unusually lucky with my original non-scientific timings.  Upon re-runs, the margin shrank, and the difference in the size of generated code (which is probably what I should've measured from the beginning) is not so large.

**Comment by e-kayrakli** [comment]
FWIW, I think we should still merge this. Some (all?) of those functions look generic, and you never know how many instantiations will be created in different codes.

**Comment by bradcray** [comment]
Thanks Engin, I was going to ask about this once people were back.  If we agree that "on principle" some or all of these routines don't obviously deserve `inline`, then I'd be in favor of checking this in and seeing what fallout there is, if any.  Put another way, I'm not likely to spend a lot more time doing manual timings before either merging or scrapping the PR.  I suppose we could also run it in the perf playground as a baby step if people felt nervous about just ripping off the band-aid.

File: modules/standard/IO.chpl [hunk 1]
```diff
@@ -2229,7 +2229,7 @@ inline proc channel.lock() throws {
 /*
    Release a channel's lock.
  */
-inline proc channel.unlock() {
+ proc channel.unlock() {
```
Comment (ronawho) [inline modules/standard/IO.chpl:2232]: Nit, but I think this has left an extra space in front of `proc` in a bunch of places.

**Comment by ronawho** [comment]
I'm also in favor of merging this.

File: modules/standard/IO.chpl (hunk 1 above)
Comment (bradcray) [reply modules/standard/IO.chpl]: Shoot, thanks for pointing that out...  Should be fixed in https://github.com/chapel-lang/chapel/pull/16887/commits/4fb95056cb7bf26e371f9ad4ea64acf79b6ef75d (but I'll take one more pass to double-check).

**Comment by bradcray** [comment]
Noting for those who, like me, look right past GitHub's cross-reference notes that this resulted in some I/O performance hits:  https://github.com/Cray/chapel-private/issues/1787

**Pull request merged**